### PR TITLE
Add Jupyter data science notebooks container

### PR DESCRIPTION
**Pull request by nathancarter**
The Jupyter project provides a [Data Science Notebooks container on DockerHub](https://hub.docker.com/r/jupyter/datascience-notebook).  It has Python, R, Julia, and all the relevant data science libraries (numpy, pandas, SciPy, tidyverse, RDatasets, much more) already installed and maintained by Jupyter.

Adding this `.devcontainer` definition to this repository just makes that container easily available to VS Code users through the "Open Folder in Container..." menu.  Given the number of people who do data science and the nice VS Code support for Jupyter notebooks, this should be a popular tool.

### CODE REVIEW

File: containers/jupyter-datascience-notebooks/.devcontainer/devcontainer.json
Comment (Chuxel) [inline containers/jupyter-datascience-notebooks/.devcontainer/devcontainer.json:3]:
```suggestion
	"name": "Jupyter Data Science Notebooks (Community)",
```
Would you mind adding a (Community) label here?

File: containers/jupyter-datascience-notebooks/.devcontainer/devcontainer.json
Comment (Chuxel) [inline containers/jupyter-datascience-notebooks/.devcontainer/devcontainer.json:1]:
```suggestion
```

This will be automatically added so you can remove it from here.

File: containers/jupyter-datascience-notebooks/README.md
Comment (Chuxel) [inline containers/jupyter-datascience-notebooks/README.md:10]:
```suggestion
# Jupyter Data Science Notebooks (Community)

## Summary

*Use Jupyter Data Science Notebooks with Python, R, Julia, and more.*

| Metadata | Value |  
|----------|-------|
| *Contributors* | Nathan Carter (@nathancarter) |
```

 This section is used by VS Code and codespaces to add a heading into drop down for Open Folder in Container / Reload Folder / Add Development Container Configuration files, so links will not work. Here's a suggested tweak. Same comment on (Community) here as well.
 
 It would also be great to include your GitHub ID so we can easily "@" you into any issues that may be raised.

**Review by Chuxel — APPROVED** [review]
This looks great!  A few readme suggestions for your approval, but otherwise this makes sense.  I assume you are okay with us looping you in to maintain the definition after its merged?

Also FYI @roblourens given #824.

**Comment by nathancarter** [comment]
Yes, I'm fine with being labeled the maintainer, although to be honest, I'm learning a lot of this as I go along!!  If you folks are around for questions, I'll help as much as I can. :)

**Comment by Chuxel** [comment]
Sounds good. Merging!

**Comment by eitsupi** [comment]
Would you like to include `common-debian.sh`? 
I have used `jupyter/scipy-notebook` (the base image of `jupyter/datascience-notebook`) in Remote-Containers, probably because `openssh-client` is not installed in the image, I couldn't `git push` from the container to the remote repository even with the ssh-agent set.

I think this integrated feature with ssh is one of the great features of VS Code Remote-Containers, so setting up with `common-debian.sh` will improve the user experience. 
https://code.visualstudio.com/docs/remote/containers#_sharing-git-credentials-with-your-container

**Comment by nathancarter** [comment]
Hi, @eitsupi.  Since this was already merged, let’s move the discussion elsewhere.  I’m not exactly sure what the best place is...maybe a new issue on this repo, tagging me in it?  When you do so, can you explain with a little more detail?  I followed the link you sent, but it did not mention `common-debian.sh`, so I’m not sure what the connection is, or where that file is.  Thanks!

**Pull request merged**
